packages/fuels-core/src/types/transaction_builders.rs

```diff
@@ -50,30 +50,48 @@ struct UnresolvedWitnessIndexes {
 #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
 pub trait BuildableTransaction: sealed::Sealed {
     type TxType: Transaction;
+    type Context;
 
-    async fn build(self, provider: impl DryRunner) -> Result<Self::TxType>;
+    async fn build(self, provider: impl DryRunner, context: Self::Context) -> Result<Self::TxType>;
+}
+
+impl sealed::Sealed for ScriptTransactionBuilder {}
 
+#[derive(Debug, Clone, Default)]
+pub enum ScriptContext {
+    /// Transaction is estimated and signatures are automatically added.
+    #[default]
+    Normal,
+    /// Transaction is estimated but no signatures are added.
     /// Building without signatures will set the witness indexes of signed coins in the
     /// order as they appear in the inputs. Multiple coins with the same owner will have
     /// the same witness index. Make sure you sign the built transaction in the expected order.
-    async fn build_without_signatures(self, provider: impl DryRunner) -> Result<Self::TxType>;
+    NoSignatures,
+    /// No estimation is done and no signatures are added. Fake coins are added if no spendable inputs
+    /// are present. Meant only for transactions that are to be dry-run with validations off.
+    /// Useful for reading state with unfunded accounts.
+    StateReadOnly,
 }
 
-impl sealed::Sealed for ScriptTransactionBuilder {}
+#[derive(Debug, Clone, Default)]
+pub enum Context {
```

Comment:
Sounds a bit too general, but I don't have an idea for a term that says "all other tx types but not script - context"

Reply:
It's now called `BuildStrategy`. Still general. works for n-1/n tx types. 

Hopefully we'll come back to this soon and redesign the builders.